File: 2239. Find Closest Number to Zero/solution.py
from typing import List

class solution:
    def findClosestNumber(self, nums:List[int]):
        closest = nums[0]
        for nums in nums:
            if abs(nums) < abs(closest):
                closest = nums
                
        if closest < 0 and abs(closest) in nums:
            return abs(closest)
        else:
            return closest
        

File: 2239. Find Closest Number to Zero/solution.java
class Solution{
    public int findClosestNumber(int[] nums){
        int closest = nums[0];
        for(int num: nums){
            if(Math.abs(num) < Math.abs(closest)){
                closest = num;
            }
        }
        if(closest < 0 && contains(nums, closest)){
            return Math.abs(closest);
        }else{
            return closest; 
        }
    }
    
    private boolean contains(int[] nums, int target){
       for(int num : nums){
            if(num == target){
                return true;
            }
        }
        return false;
    }
}